src/webrtc/congestion_control/cc.md

```diff
@@ -0,0 +1,164 @@
+> the available bandwidth estimate should stay within a given bound.
+
+The bandwidth estimation provided by the congestion controller is in fact a request to the sender for producing 
+the stream with a specific bitrate.
+If the network behavior remains unchanged after increasing the sender rate the estimation is increased.
+If the network starts being congested (transmission delay increases, packets get lost), the estimation
+is decreased.
+This way we are slowly increasing the sender rate up to the point where the network can't accept more data.
+
+There are two increase modes: multiplicative and additive.
+In the multiplicative mode, the estimation is increased by no more than 8% per second.
+In the additvie mode, the estimation is increased with at most half a packet per `response_time_interval` (RTT + 100ms).
+In other words, the additive mode is much slower and is used when the estimation is close to convergence.
+
+When estimation is decreased, it is set to 85%-95% (exact value depends on the implementation) of the latest receiver rate.
+
+The whole process is shown in the following figure.
+
+![GCC standard flow](./gcc_std_flow.png)
+
+One can use the fact, that in the decrease state estimation is set to 85& of the latest receiver rate and 
+set sender rate more aggressively which should result in faster convergence.
+
+![GCC overuse flow](./gcc_overuse_flow.png)
+
+## GCC & Simulcast
+
+Let's consider a scenario where the network link between the server and the receiver is quite poor.
+In such case, the sender will send high-quality video to the server but the receiver won't be able to receive it.
+Limiting media quality on the server side is challenging as sending video with a rate lower than the video
+bitrate will result in frequent video rebuffering (video freezes).
+What the server can do is request the sender to e.g. increase its compression level, decrease resolution or FPS,
+or reencode the video on its own.
+The former is problematic in sessions with more than 2 participants as there might be e.g. only one host that
+cannot handle high-quality video but the server will send all of the participants lower resolution.
+The latter requires a lot of resources and is hard to be achieved in real-time.
+This leads us to the simulcast where the sender sends multiple qualities in multiple FPS and the server decides
+which quality should be sent to which participant.
+There are still some bitrate levels so the server cannot lower its rate to any level but in most cases,
+there is always some quality that will fit into the receiver network condition.
+
+![simulcast overview](./simulcast.png)
+
+As stated before, bandwidth estimation depends on the amount of data that is sent therefore, if the server sends
+little data the estimation will be low.
+To recover from this situation, the server has to increase the amount of data it sends which leads us to a connection probing.
```

Comment:
I think we should add something between these two.
```suggestion
As stated before, bandwidth estimation depends on the amount of data that is sent therefore, if the server sends little data the estimation will be low.
An implication of this fact is. assuming that SFU selects the highest simulcast layer that fits the estimation, the receiver will never be able to get a layer higher than the one they're currently getting.
When you take varying network conditions into account and the fact that in the decreasing mode the estimation is set to only 85% of the sender rate, which is equal to the bitrate of the currently selected layer, we found that all receivers tend to land on the lowest possible layer pretty quickly.
To recover from this situation, the server has to increase the amount of data it sends which leads us to a connection probing.
```

Reply:
> An implication of this fact is. assuming that SFU selects the highest simulcast layer that fits the estimation, the receiver will never be able to get a layer higher than the one they're currently getting.


That's not fully correct. Imagine we are sending 1.5Mbps. The max estimation is 1.5 * 1.5Mbps which is ~2.3Mbps. So we have about 750kbps free to use. This might be enough to increase at least temporal layer.

>the fact that in the decreasing mode the estimation is set to only 85% of the sender rate, which is equal to the bitrate of the currently selected layer,

It's set to 85% of the receiver rate which doesn't need to be equal to the bitrate of currently selected layer